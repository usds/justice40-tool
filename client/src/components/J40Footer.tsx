import React from 'react';
import {
  Address,
  Logo,
} from '@trussworks/react-uswds';
import {NavList} from '@trussworks/react-uswds';
import {useIntl} from 'gatsby-plugin-intl';

import J40MainGridContainer from './J40MainGridContainer';

// @ts-ignore
import whitehouseIcon from '../images/eop-seal.svg';
import * as COMMON_COPY from '../data/copy/common';

const J40Footer = () => {
  const intl = useIntl();

  const NAVLINKS = [
    ['Contact',
      <Address
        key={'footeraddress'}
        className={'j40-footer-address'}
        size={'big'}
        items={[
          COMMON_COPY.FOOTER_CEQ_ADDRESS.NAME,
          COMMON_COPY.FOOTER_CEQ_ADDRESS.STREET,
          COMMON_COPY.FOOTER_CEQ_ADDRESS.CITY_STATE,
          COMMON_COPY.FOOTER_CEQ_ADDRESS.PHONE,
        ]}
      />,
    ],
    [
      intl.formatMessage(COMMON_COPY.FOOTER.MORE_INFO),
      <a
        className={'footer-link-first-child'}
        key={'whitehouselink2'}
        href={'https://www.whitehouse.gov/'}
        target={'_blank'}
        rel={'noreferrer'}>Whitehouse.gov</a>,
      <a
        key="foialink"
        target={'_blank'}
        rel={'noreferrer'}
        href={'https://www.whitehouse.gov/ceq/foia'}>
        {intl.formatMessage(COMMON_COPY.FOOTER.FOIA)}
      </a>,
      <a
        key={'privacylink'}
        target={'_blank'}
        rel={'noreferrer'}
        href={'https://www.whitehouse.gov/privacy/'}>
        {intl.formatMessage(COMMON_COPY.FOOTER.PRIVACY)}
      </a>,
    ],
    [
      intl.formatMessage(COMMON_COPY.FOOTER.QUESTIONS),
      <a
        className={'footer-link-first-child'}
        key={'contactlink'}
        href={'https://www.usa.gov/'}>
        {intl.formatMessage(COMMON_COPY.FOOTER.CONTACT_LINK)}
      </a>,
    ],
  ];

  // see https://designsystem.digital.gov/components/footer/
  return (
    // we cannot use trussworks Footer because it doesn't layout correct
    // and there's no easy way to override. It comes down to the
    // `className="mobile-lg:grid-col-6 desktop:grid-col-3">` needs to be
    // `className="mobile-lg:grid-col-12 desktop:grid-col-4">` ugh.
    <footer className={'j40-footer'}>
      <div className="usa-footer__primary-section pb">
        <J40MainGridContainer>
          <div className={'grid-row grid-gap-4 padding-bottom-6 tablet-lg:grid-col4'}>
            {NAVLINKS.map((links, i) => (
              <div key={`linkSection-${i}`}
                className="mobile-lg:grid-col-12 desktop:grid-col-4">
                <NavSection links={links} />
              </div>
            ))}
          </div>
        </J40MainGridContainer>
      </div>

      <div className="usa-footer__secondary-section">
        <J40MainGridContainer>
          <Logo
            size="medium"
            key={'logoimg'}
            className={'j40-footer-logo'}
            image={
              <img
                className={'usa-footer__logo-img'}
                src={whitehouseIcon}
                alt={intl.formatMessage(COMMON_COPY.FOOTER.LOGO_ALT)}/>
            }
            heading={<h5>{intl.formatMessage(COMMON_COPY.FOOTER.TITLE)}</h5>}
          />
        </J40MainGridContainer>
      </div>
    </footer>
  );
};

const NavSection = ({
  links,
}: {
  links: React.ReactNode[]
}): React.ReactElement => {
  const [primaryLinkOrHeading, ...secondaryLinks] = links;
  return (
    <section className={'usa-footer__primary-content'}>
      <h4 className="padding-top-1 padding-bottom-0">{primaryLinkOrHeading}</h4>
      <NavList className={'padding-bottom-4'} type="footerSecondary" items={secondaryLinks} />
    </section>
  );
};

export default J40Footer;
