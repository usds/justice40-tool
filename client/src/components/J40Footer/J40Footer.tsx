import React from 'react';
import {
  Address,
  Logo,
} from '@trussworks/react-uswds';
import {NavList} from '@trussworks/react-uswds';
import {useIntl} from 'gatsby-plugin-intl';

import {hyphenizeString} from '../../../cypress/integration/common/helpers';
import J40MainGridContainer from '../J40MainGridContainer';
import LinkTypeWrapper from '../LinkTypeWrapper';
import SurveyButton from '../SurveyButton';

// @ts-ignore
import whitehouseIcon from '../../images/eop-seal.svg';
import * as COMMON_COPY from '../../data/copy/common';

const J40Footer = () => {
  const intl = useIntl();

  const NAVLINKS = [
    [
      intl.formatMessage(COMMON_COPY.FOOTER.CONTACT),
      <Address
        className={'j40-footer-address'}
        key={'footeraddress'}
        size={'big'}
        items={[
          COMMON_COPY.FOOTER_CEQ_ADDRESS.NAME,
          COMMON_COPY.FOOTER_CEQ_ADDRESS.STREET,
          COMMON_COPY.FOOTER_CEQ_ADDRESS.CITY_STATE,
          COMMON_COPY.FOOTER_CEQ_ADDRESS.PHONE,
        ]}
      />,
      <LinkTypeWrapper
        linkText={intl.formatMessage(COMMON_COPY.FOOTER.SIGN_UP)}
        internal={false}
        url={COMMON_COPY.FOOTER.SIGN_UP_LINK}
        openUrlNewTab={true}
        key={'signup'}
        dataCy={hyphenizeString(COMMON_COPY.FOOTER.SIGN_UP.defaultMessage)}
      />,
    ],
    [
      intl.formatMessage(COMMON_COPY.FOOTER.MORE_INFO),
      <LinkTypeWrapper
        className={'footer-link-first-child'}
        linkText={intl.formatMessage(COMMON_COPY.FOOTER.WHITEHOUSE)}
        internal={false}
        url={intl.formatMessage(COMMON_COPY.FOOTER.WHITEHOUSE_LINK)}
        openUrlNewTab={true}
        key={'whitehouselink2'}
        dataCy={hyphenizeString(COMMON_COPY.FOOTER.WHITEHOUSE.defaultMessage)}
      />,
      <LinkTypeWrapper
        linkText={intl.formatMessage(COMMON_COPY.FOOTER.FOIA)}
        internal={false}
        url={'https://www.whitehouse.gov/ceq/foia'}
        openUrlNewTab={true}
        key={'foialink'}
        dataCy={hyphenizeString(COMMON_COPY.FOOTER.FOIA.defaultMessage)}
      />,
      <LinkTypeWrapper
        linkText={intl.formatMessage(COMMON_COPY.FOOTER.PRIVACY)}
        internal={false}
        url={intl.formatMessage(COMMON_COPY.FOOTER.PRIVACY_LINK)}
        openUrlNewTab={true}
        key={'privacylink'}
        dataCy={hyphenizeString(COMMON_COPY.FOOTER.PRIVACY.defaultMessage)}
      />,
      <LinkTypeWrapper
        linkText={intl.formatMessage(COMMON_COPY.FOOTER.FIND_CONTACT)}
        internal={false}
        url={intl.formatMessage(COMMON_COPY.FOOTER.FIND_CONTACT_LINK)}
        openUrlNewTab={true}
        key={'contactlink'}
        dataCy={hyphenizeString(COMMON_COPY.FOOTER.FIND_CONTACT.defaultMessage)}
      />,
    ],
    [
      intl.formatMessage(COMMON_COPY.FOOTER.CONTRIBUTE),
      <LinkTypeWrapper
        linkText={intl.formatMessage(COMMON_COPY.FOOTER.GITHUB_LINK_TEXT)}
        internal={false}
        url={intl.formatMessage(COMMON_COPY.FOOTER.GITHUB_LINK)}
        openUrlNewTab={true}
        className={'footer-link-first-child'}
        key={'contactlink'}
        dataCy={hyphenizeString(COMMON_COPY.FOOTER.GITHUB_LINK_TEXT.defaultMessage)}
      />,
    ],
  ];

  // see https://designsystem.digital.gov/components/footer/
  return (
    // we cannot use trussworks Footer because it doesn't layout correct
    // and there's no easy way to override. It comes down to the
    // `className="mobile-lg:grid-col-6 desktop:grid-col-3">` needs to be
    // `className="mobile-lg:grid-col-12 desktop:grid-col-4">` ugh.
    <footer className={'j40-footer'}>
      <div className="usa-footer__primary-section pb2" data-cy={`footer-primary-block`}>
        <J40MainGridContainer>
          <div className={'grid-row tablet-lg:grid-col4'}>
            {NAVLINKS.map((links, i) => (
              <div
                key={`linkSection-${i}`}
                className="mobile-lg:grid-col-12 desktop:grid-col-4">
                <NavSection links={links} />
              </div>
            ))}
          </div>
        </J40MainGridContainer>
      </div>

      <div className="usa-footer__secondary-section">
        <J40MainGridContainer>
          <Logo
            size="medium"
            key={'logoimg'}
            image={
              <img
                className={'usa-footer__logo-img'}
                src={whitehouseIcon}
                alt={intl.formatMessage(COMMON_COPY.FOOTER.LOGO_ALT)}/>
            }
            heading={
              <div className={'j40-footer-ceq-font'}>
                {intl.formatMessage(COMMON_COPY.FOOTER.TITLE)}
              </div>
            }
          />
        </J40MainGridContainer>
      </div>
      <SurveyButton />
    </footer>
  );
};

const NavSection = ({
  links,
}: {
  links: React.ReactNode[]
}): React.ReactElement => {
  const [primaryLinkOrHeading, ...secondaryLinks] = links;
  return (
    <section>
      <div className="j40-h4">{primaryLinkOrHeading}</div>
      <NavList type="footerSecondary" items={secondaryLinks} />
    </section>
  );
};

export default J40Footer;
