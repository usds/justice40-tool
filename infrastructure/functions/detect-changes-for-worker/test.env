REGION=us-east-1
STAGE=sit

# ESC Cluster name that will from the containers
ECS_CLUSTER=j40-sit-justice40-data-harvester-ECSCluster-ktXGGU9zjwkb

# VPC Private Subnet that has a NAT GAteway
VPC_SUBNET_ID=subnet-07e68cb57322f7b1f

# Names of the container and task names.
GDAL_TASK_DEFINITION=sit-justice40-data-harvester-gdal
GDAL_CONTAINER_DEFINITION=sit-justice40-data-harvester-osgeo-gdal
TIPPECANOE_TASK_DEFINITION=sit-justice40-data-harvester-tippecanoe
TIPPECANOE_CONTAINER_DEFINITION=sit-justice40-data-harvester-mgiddens-tippecanoe

# AWS Credentials
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1