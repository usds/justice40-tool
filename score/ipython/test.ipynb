{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1a4c0c68",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "70b3a793",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/fips_states.csv')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c514aad8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fips</th>\n",
       "      <th>state_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Alabama</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Alaska</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>Arizona</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>Arkansas</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>California</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fips  state_name\n",
       "0     1    Alabama \n",
       "1     2     Alaska \n",
       "2     4    Arizona \n",
       "3     5   Arkansas \n",
       "4     6  California"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9ee44d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
