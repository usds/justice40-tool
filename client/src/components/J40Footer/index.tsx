import J40Footer from './J40Footer';

export default J40Footer;
