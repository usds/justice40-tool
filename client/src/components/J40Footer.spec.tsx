import React from 'react';
import renderer from 'react-test-renderer';
import J40Footer from './J40Footer';
import {LocalizedComponent} from '../test/testHelpers';

describe('J40Footer', () => {
  it('renders correctly', () => {
    const tree = renderer
        .create(<LocalizedComponent><J40Footer /></LocalizedComponent>)
        .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
