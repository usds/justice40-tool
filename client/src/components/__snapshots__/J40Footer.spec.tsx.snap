// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`J40Footer renders correctly 1`] = `
<DocumentFragment>
  <footer
    class="usa-footer usa-footer--big j40-footer"
  >
    <div
      class="usa-footer__primary-section"
    >
      <div
        class="grid-container"
        data-testid="gridContainer"
      >
        <div
          class=""
          data-testid="grid"
        >
          <nav
            aria-label="Footer navigation"
            class="usa-footer__nav"
          >
            <div
              class="grid-row grid-gap-4"
            >
              <div
                class="mobile-lg:grid-col-6 desktop:grid-col-3"
              >
                <section
                  class="usa-footer__primary-content usa-footer__primary-content--collapsible"
                >
                  <h4
                    class="usa-footer__primary-link"
                  >
                    Contacts
                  </h4>
                  <ul
                    class="usa-list usa-list--unstyled"
                  >
                    <li
                      class="usa-footer__secondary-link"
                    >
                      <address
                        class="usa-footer__address j40-footer-address"
                      >
                        <div
                          class="usa-footer__contact-info grid-row grid-gap"
                        >
                          <div
                            class="grid-col-auto"
                          >
                            730 Jackson Pl NW
                          </div>
                          <div
                            class="grid-col-auto"
                          >
                            Washington, D.C. 20506
                          </div>
                          <div
                            class="grid-col-auto"
                          >
                            (202) 395-5750
                          </div>
                        </div>
                      </address>
                    </li>
                  </ul>
                </section>
              </div>
              <div
                class="mobile-lg:grid-col-6 desktop:grid-col-3"
              >
                <section
                  class="usa-footer__primary-content usa-footer__primary-content--collapsible"
                >
                  <h4
                    class="usa-footer__primary-link"
                  >
                    More Information
                  </h4>
                  <ul
                    class="usa-list usa-list--unstyled"
                  >
                    <li
                      class="usa-footer__secondary-link"
                    >
                      <a
                        href="https://www.whitehouse.gov/"
                        rel="noreferrer"
                        target="_blank"
                      >
                        Whitehouse.gov
                      </a>
                    </li>
                    <li
                      class="usa-footer__secondary-link"
                    >
                      <a
                        href="https://www.whitehouse.gov/ceq/foia"
                        rel="noreferrer"
                        target="_blank"
                      >
                        Freedom of Information Act (FOIA)
                      </a>
                    </li>
                    <li
                      class="usa-footer__secondary-link"
                    >
                      <a
                        href="https://www.whitehouse.gov/privacy/"
                        rel="noreferrer"
                        target="_blank"
                      >
                        Privacy Policy
                      </a>
                    </li>
                  </ul>
                </section>
              </div>
              <div
                class="mobile-lg:grid-col-6 desktop:grid-col-3"
              >
                <section
                  class="usa-footer__primary-content usa-footer__primary-content--collapsible"
                >
                  <h4
                    class="usa-footer__primary-link"
                  >
                    Have a question about government services?
                  </h4>
                  <ul
                    class="usa-list usa-list--unstyled"
                  >
                    <li
                      class="usa-footer__secondary-link"
                    >
                      <a
                        href="https://www.usa.gov/"
                      >
                        Find a contact at USA.gov
                      </a>
                    </li>
                  </ul>
                </section>
              </div>
            </div>
          </nav>
        </div>
      </div>
    </div>
    <div
      class="usa-footer__secondary-section"
    >
      <div
        class="grid-container"
      >
        <div
          class="usa-footer__logo grid-row mobile-lg:grid-col-6 mobile-lg:grid-gap-2 j40-footer-logo"
          data-testid="footerLogo"
        >
          <div
            class="mobile-lg:grid-col-auto"
          >
            <img
              alt="Whitehouse logo"
              class="usa-footer__logo-img"
              src="test-file-stub"
            />
          </div>
          <div
            class="mobile-lg:grid-col-auto"
          >
            <p
              class="j40-footer-logo-heading"
            >
              Council on Environmental Quality
            </p>
          </div>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
